session.go

```diff
@@ -1,97 +1,77 @@
 package bond
 
 import (
+	"database/sql"
+	"fmt"
 	"reflect"
 	"sync"
 
-	"upper.io/db"
+	"upper.io/db.v2"
+	"upper.io/db.v2/lib"
 )
 
 type Session interface {
-	db.Tx
+	lib.SQLDatabase
```

Comment:
@xiam that makes sense for sure. check out https://github.com/upper/bond/pull/20#issuecomment-231083823 .. in this design, if Session interface implements db.TxDatabase(), this wouldn't have any Commit() or Rollback() methods.. it would just have methods to make a NewTx. The one method UseTx(Tx) would be okay too I think because it would just clone the session and use the tx that was passed. I think thats okay.. 

as well, the design in my comment renames BondTx(func) to just Tx(func)